bitcoin/src/bip32.rs

```diff
@@ -256,94 +365,229 @@ impl serde::Serialize for ChildNumber {
         u32::from(*self).serialize(serializer)
     }
 }
+/// normal_derivation_path module with inner methods
+pub mod normal_derivation_path {
+    use super::*;
 
-/// Trait that allows possibly failable conversion from a type into a
-/// derivation path
-pub trait IntoDerivationPath {
-    /// Converts a given type into a [`DerivationPath`] with possible error
-    fn into_derivation_path(self) -> Result<DerivationPath, Error>;
-}
+    ///A DerivationPath that can only contains NormalChildNumbers (i.e. no hardened indexes)
+    #[derive(Clone, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
+    pub struct NormalDerivationPath(Vec<NormalChildNumber>);
 
-/// A BIP-32 derivation path.
-#[derive(Clone, PartialEq, Eq, Ord, PartialOrd, Hash)]
-pub struct DerivationPath(Vec<ChildNumber>);
+    #[cfg(feature = "serde")]
+    internals::serde_string_impl!(
+        NormalDerivationPath,
+        "a BIP-32 derivation path containing normal numbers only"
+    );
+
+    impl NormalDerivationPath {
+        /// Create a new `NormalDerivationPath` from a vector of `NormalChildNumber`
+        pub fn new(vec: Vec<NormalChildNumber>) -> Self { NormalDerivationPath(vec) }
+        /// Returns length of the `NormalDerivationPath`
+        pub fn len(&self) -> usize { self.0.len() }
```

Comment:
Weird that `clippy` isn't complaining about missing `is_empty`. Something is wrong with our linting.

Reply:
So you want a `is_empy()` method ?

Reply:
Hmm, it should be called `is_master` I guess.

Reply:
We already have this function